common/Resources/ti.internal/extensions/js/console.js

```diff
@@ -1,30 +1,156 @@
-import { formatWithOptions } from '../node/internal/util/inspect';
+import { formatWithOptions, inspect } from '../node/internal/util/inspect';
 
-const nativeDebug = console.debug;
-const nativeError = console.error;
-const nativeInfo = console.info;
-const nativeLog = console.log;
-const nativeWarn = console.warn;
+function logTime(self, label, logData) {
+	label = `${label}`;
+	const startTime = self._times.get(label);
+	if (!startTime) {
+		process.emitWarning(`Label "${label}" does not exist`);
+		return true;
+	}
+	const duration = Date.now() - startTime;
+	if (logData) {
+		self.log(`${label}: ${duration}ms`, ...logData);
+	} else {
+		self.log(`${label}: ${duration}ms`);
+	}
+	return false;
+}
 
 const kColorInspectOptions = { colors: true };
 const kNoColorInspectOptions = {};
 
-console.debug = function (...args) {
-	nativeDebug.call(console, formatWithOptions(kColorInspectOptions, ...args));
-};
+class Console {
+	constructor(options) {
+		// Allow passing in a fake Ti.API that we can hijack the info/warn/error/debug calls!
+		if (options && options.apiName === 'Ti.API') {
+			this._apiModule = options;
+		} else {
+			// if (!options || typeof options.write === 'function') {
```

Comment:
It might be good to spit out a warning / error here since this will practically create a broken `console` object which will lead to a crash when used due to `_apiModule` being undefined.